staging/src/k8s.io/apiserver/pkg/endpoints/discovery/aggregated/handler.go

```diff
@@ -505,11 +512,25 @@ func (rdm *resourceDiscoveryManager) serveHTTP(resp http.ResponseWriter, req *ht
 	response := cache.cachedResponse
 	etag := cache.cachedResponseETag
```

Comment:
Can you write up an explanation of why the  etag doesn't vary by requested encoding?

Reply:
Etag will be the same but the Vary header will prevent the same cache from being used if a different encoding is requested.

https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apiserver/pkg/endpoints/discovery/aggregated/etag.go#L50

Reply:
Is there risk that a client might use tags as coordinates (e.g. keys in a map) and accidentally collide different encodings on the client side?

Reply:
Can you elaborate on tags as coordinates?

Reply:
Would a client every, say, want to cache the data and choose to use the tags as the hash keys?  If so, is there a risk they could have collisions for the same etag but different encoding?

Reply:
Vary is a standard HTTP header part of [RFC7231](https://www.rfc-editor.org/rfc/rfc7231.html#section-7.1.4)

Also "When a cache receives a request that can be satisfied by a stored response that has a [Vary](https://httpwg.org/specs/rfc7231.html#header.vary) header field ([Section 7.1.4](https://httpwg.org/specs/rfc7231.html#header.vary) of [[RFC7231]](https://httpwg.org/specs/rfc7234.html#RFC7231)), it MUST NOT use that response unless all of the selecting header fields nominated by the Vary header field match in both the original request (i.e., that associated with the stored response), and the presented request."

I guess in theory a client can choose to ignore it but standard clients should respect it? For instance the golang http cached client specifically looks for this keyword when deciding whether to use the cache.

Reply:
Can I get a summary of this as a comment in the code for future-us?